src/main/java/de/retest/surili/commons/ga/TestLengthFitness.java

```diff
@@ -18,13 +19,25 @@
 @Value
 public class TestLengthFitness implements EfficiencyFitness {
 
-	@NonNull
-	TestSuite testSuite;
+	private final int length;
+
+	public TestLengthFitness( final TestSuite testSuite ) {
+		this( testSuite.getTestCases() );
+	}
+
+	public TestLengthFitness( final Iterable<TestCase> testCases ) {
+		this( StreamSupport.stream( testCases.spliterator(), false ) //
+				.mapToInt( TestCase::getSize ) //
+				.sum() );
+	}
+
+	public TestLengthFitness( final int length ) {
+		this.length = length;
+	}
 
 	@Override
 	public double getFitness() {
-		return -1 * testSuite.getTestCases().stream() //
-				.mapToInt( TestCase::getSize ) //
-				.sum();
+		return -1 * length;
```

Comment:
Well, I could think about immediately calculating the actual fitness, but a) it's done similar to this elsewhere and b) I don't think negating an integer (i.e. setting a bit) isn't that expensive. 😉 To give more context: To evaluate the fitness, we have to execute the altered individual; that is, execute a possibly mutated and recombinated test suite via Selenium. So, I think we can safely I assume that this is the least expensive part.

Reply:
Sure, retrieving the length (or even negating it) is not that expensive. I just wanted to point out, that you might want to consider where the fitness is calculated: In the constructor or in the `getFitness()`.

As you said. evaluating fitness involves executing the individual. That's why it may as well be in the constructor as it is part of the fitness result. However, you could also define a `ExecuteTestSuiteFitness` which does the actual said step above in `getFitness`. It just depends on the framework you are using and to keep it consistent.